docs/book/driver-deployment/upgrade.md

```diff
@@ -21,12 +89,77 @@ kubectl delete -f https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-
 
 Wait for vSphere CSI Controller Pod and vSphere CSI Nodes Pods to be completely deleted.
 
-Install `v2.0.1` driver
+### Install `v2.3.0` driver
 
-```bash
-kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-driver/release-2.0/manifests/v2.0.1/vsphere-67u3/rbac/vsphere-csi-controller-rbac.yaml
-kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-driver/release-2.0/manifests/v2.0.1/vsphere-67u3/deploy/vsphere-csi-controller-deployment.yaml
-kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-driver/release-2.0/manifests/v2.0.1/vsphere-67u3/deploy/vsphere-csi-node-ds.yaml
-```
+Refer to the [release notes](../releases/v2.3.0.md) of `v2.3.0` for detail.
+
+#### Pre-requiresites for `v2.3.0`
+
+`v2.3.0` release of the driver requires DNS forwarding configuration in CoreDNS ConfigMap to help resolve vSAN file share hostname.
+
+Modify the CoreDNS ConfigMap and add the conditional forwarder configuration:
+
+   ```bash
+   kubectl -n kube-system edit configmap coredns
+   ```
+
+Output:
+
+   ```bash
+   .:53 {
+    errors
+    health {
+       lameduck 5s
+    }
+    ready
+    kubernetes cluster.local in-addr.arpa ip6.arpa {
+       pods insecure
+       fallthrough in-addr.arpa ip6.arpa
+       ttl 30
+    }
+    prometheus :9153
+    forward . /etc/resolv.conf {
+       max_concurrent 1000
+    }
+    cache 30
+    loop
+    reload
+    loadbalance
+   }
+   vsanfs-sh.prv:53 {
+   errors
+   cache 30
+   forward . 10.161.191.241
+   }
+   ```
+
+In the above configuration `vsanfs-sh.prv` is the DNS suffixes for vSAN file Service and `10.161.191.241` is the DNS server that helps resolve file share hostname.
+DNS suffixes and DNS IP address can be obtained from vCenter (vSphere Cluster -> Configure -> vSAN -> Services -> File Service)
+
+#### Install v2.3.0 release of the driver
+
+1. Create a new namespace for vSphere CSI Driver.
+
+   ```bash
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-driver/v2.3.0-rc.5/manifests/vanilla/namespace.yaml
+   ```
+
+2. Copy `vsphere-config-secret` secret from `kube-system` namespace to the new namespace `vmware-system-csi`.
+
+   ```bash
+   kubectl get secret vsphere-config-secret --namespace=kube-system -o yaml | sed 's/namespace: .*/namespace: vmware-system-csi/' | kubectl apply -f -
+   ```
+
+3. Delete `vsphere-config-secret` secret from `kube-system` namespace.
+
+   ```bash
+   kubectl delete secret vsphere-config-secret --namespace=kube-system
+   ```
+
+4. Install Driver
+
+   ```bash
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/vsphere-csi-driver/v2.3.0-rc.5/manifests/vanilla/vsphere-csi-driver.yaml
```

Comment:
Do we plan to change `v2.3.0-rc.5` to `v2.3.0` later?

Reply:
it should be `v2.3.0`. fixed.